File: backend/internal/handler/profanity.go
package handler

import "strings"

var profanityMap = []struct {
	word        string
	replacement string
}{
	{"시발", "시**"},
	{"씨발", "씨**"},
	{"좆", "*"},
	{"존나", "존**"},
	{"지랄", "지**"},
	{"개소리", "개**"},
	{"병신", "병**"},
	{"새끼", "새**"},
	{"개새", "개**"},
	{"미친놈", "미**"},
	{"미친년", "미**"},
}

func MaskProfanity(text string) string {
	result := text
	for _, p := range profanityMap {
		result = strings.ReplaceAll(result, p.word, p.replacement)
	}
	return result
}

File: frontend/lib/utils/profanity.ts
// 욕설 마스킹 — 프론트엔드 전체 적용
const PROFANITY_MAP: [RegExp, string][] = [
  [/시발/g, '시**'],
  [/씨발/g, '씨**'],
  [/씹/g, '*'],
  [/좆/g, '*'],
  [/존나/g, '존**'],
  [/지랄/g, '지**'],
  [/개소리/g, '개**'],
  [/병신/g, '병**'],
  [/새끼/g, '새**'],
  [/ㅅㅂ/g, '**'],
  [/ㅈㄴ/g, '**'],
  [/ㅂㅅ/g, '**'],
  [/개새/g, '개**'],
  [/미친놈/g, '미**'],
  [/미친년/g, '미**'],
  [/꺼져/g, '꺼**'],
  [/닥쳐/g, '닥**'],
]

export function maskProfanity(text: string): string {
  let result = text
  for (const [pattern, replacement] of PROFANITY_MAP) {
    result = result.replace(pattern, replacement)
  }
  return result
}
